### PR TITLE
add parallel_spec rake task

### PR DESCRIPTION
**Pull request by jlambert121**

### CODE REVIEW

**Comment by DavidS** [comment]
I would like to merge that, but trying it out with a module I had laying around, in the spec_clean phase it tries to run another `rspec` command. Do you have any idea where this could be coming from?

```
$ time bundle exec rake -t parallel_spec
[...]
Finished in 1 minute 14.48 seconds (files took 1.12 seconds to load)
149 examples, 0 failures


1311 examples, 0 failures

Took 78 seconds (1:18)
** Invoke spec_clean (first_time)
** Execute spec_clean
/usr/bin/ruby2.3 -I/home/david/gems/ruby/2.3.0/gems/rspec-core-3.4.4/lib:/home/david/gems/ruby/2.3.0/gems/rspec-support-3.4.1/lib /home/david/gems/ruby/2.3.0/gems/rspec-core-3.4.4/exe/rspec --pattern spec/\*\*\{,/\*/\*\*\}/\*_spec.rb
/home/david/gems/ruby/2.3.0/gems/beaker-rspec-5.3.0/lib/beaker-rspec/helpers/serverspec.rb:43: warning: already initialized constant Module::VALID_OPTIONS_KEYS
/home/david/gems/ruby/2.3.0/gems/specinfra-2.56.0/lib/specinfra/configuration.rb:4: warning: previous definition of VALID_OPTIONS_KEYS was here
Hypervisor for centos-66-x64 is vagrant
Beaker::Hypervisor, found some vagrant boxes to create
/home/david/gems/ruby/2.3.0/gems/beaker-2.38.1/lib/beaker/hypervisor/vagrant.rb:196:in `block (2 levels) in vagrant_cmd': Failed to exec 'vagrant destroy --force'. Error was /usr/lib/ruby/vendor_ruby/bundler/rubygems_integration.rb:304:in `block in replace_gem': vagrant-wrapper is not part of the bundle. Add it to Gemfile. (Gem::LoadError) (RuntimeError)
	from /home/david/gems/ruby/2.3.0/bin/vagrant:22:in `<main>'
	from /usr/lib/ruby/2.3.0/open3.rb:205:in `popen_run'
	from /usr/lib/ruby/2.3.0/open3.rb:95:in `popen3'
	from /home/david/gems/ruby/2.3.0/gems/beaker-2.38.1/lib/beaker/hypervisor/vagrant.rb:191:in `block in vagrant_cmd'
	from /home/david/gems/ruby/2.3.0/gems/beaker-2.38.1/lib/beaker/hypervisor/vagrant.rb:189:in `chdir'
	from /home/david/gems/ruby/2.3.0/gems/beaker-2.38.1/lib/beaker/hypervisor/vagrant.rb:189:in `vagrant_cmd'
	from /home/david/gems/ruby/2.3.0/gems/beaker-2.38.1/lib/beaker/hypervisor/vagrant.rb:147:in `provision'
	from /home/david/gems/ruby/2.3.0/gems/beaker-2.38.1/lib/beaker/hypervisor.rb:75:in `create'
	from /home/david/gems/ruby/2.3.0/gems/beaker-2.38.1/lib/beaker/network_manager.rb:69:in `block in provision'
	from /home/david/gems/ruby/2.3.0/gems/beaker-2.38.1/lib/beaker/network_manager.rb:68:in `each_key'
	from /home/david/gems/ruby/2.3.0/gems/beaker-2.38.1/lib/beaker/network_manager.rb:68:in `provision'
	from /home/david/gems/ruby/2.3.0/gems/beaker-rspec-5.3.0/lib/beaker-rspec/beaker_shim.rb:35:in `provision'
	from /home/david/gems/ruby/2.3.0/gems/beaker-rspec-5.3.0/lib/beaker-rspec/spec_helper.rb:46:in `block in <top (required)>'
	from /home/david/gems/ruby/2.3.0/gems/rspec-core-3.4.4/lib/rspec/core.rb:97:in `configure'
	from /home/david/gems/ruby/2.3.0/gems/beaker-rspec-5.3.0/lib/beaker-rspec/spec_helper.rb:5:in `<top (required)>'
	from /home/david/git/puppetlabs-apache/spec/spec_helper_acceptance.rb:1:in `require'
	from /home/david/git/puppetlabs-apache/spec/spec_helper_acceptance.rb:1:in `<top (required)>'
	from /home/david/git/puppetlabs-apache/spec/acceptance/apache_parameters_spec.rb:1:in `require'
	from /home/david/git/puppetlabs-apache/spec/acceptance/apache_parameters_spec.rb:1:in `<top (required)>'
	from /home/david/gems/ruby/2.3.0/gems/rspec-core-3.4.4/lib/rspec/core/configuration.rb:1361:in `load'
	from /home/david/gems/ruby/2.3.0/gems/rspec-core-3.4.4/lib/rspec/core/configuration.rb:1361:in `block in load_spec_files'
	from /home/david/gems/ruby/2.3.0/gems/rspec-core-3.4.4/lib/rspec/core/configuration.rb:1359:in `each'
	from /home/david/gems/ruby/2.3.0/gems/rspec-core-3.4.4/lib/rspec/core/configuration.rb:1359:in `load_spec_files'
	from /home/david/gems/ruby/2.3.0/gems/rspec-core-3.4.4/lib/rspec/core/runner.rb:106:in `setup'
	from /home/david/gems/ruby/2.3.0/gems/rspec-core-3.4.4/lib/rspec/core/runner.rb:92:in `run'
	from /home/david/gems/ruby/2.3.0/gems/rspec-core-3.4.4/lib/rspec/core/runner.rb:78:in `run'
	from /home/david/gems/ruby/2.3.0/gems/rspec-core-3.4.4/lib/rspec/core/runner.rb:45:in `invoke'
	from /home/david/gems/ruby/2.3.0/gems/rspec-core-3.4.4/exe/rspec:4:in `<main>'
/usr/bin/ruby2.3 -I/home/david/gems/ruby/2.3.0/gems/rspec-core-3.4.4/lib:/home/david/gems/ruby/2.3.0/gems/rspec-support-3.4.1/lib /home/david/gems/ruby/2.3.0/gems/rspec-core-3.4.4/exe/rspec --pattern spec/\*\*\{,/\*/\*\*\}/\*_spec.rb failed

real	1m23.933s
user	7m55.332s
sys	0m8.212s
$
```

**Comment by jlambert121** [comment]
Off the top of my head, no, haven't seen this before, I'll have to look into it some more.

File: Gemfile [hunk 1]
```diff
@@ -6,6 +6,7 @@ gem 'rubocop'
 gem 'puppet-lint'
 gem 'puppet-syntax'
 gem 'mocha'
+gem 'parallel_tests'
```
Comment (DavidS) [inline Gemfile]: This needs to go to the .gemspec, not here.

**Comment by jlambert121** [comment]
Updated

**Comment by alexharv074** [comment]
Ah ok, I just raised a PR to do more or less the same thing, I didn't see this one.

**Comment by alexharv074** [comment]
My PR is [here](https://github.com/puppetlabs/puppetlabs_spec_helper/pull/152).

**Comment by DavidS** [comment]
Merged #152 with more docs.

**Pull request closed without merging**